sdk/core/logger/src/logger.ts

```diff
@@ -113,17 +115,18 @@ export function createClientLogger(namespace: string): AzureLogger {
   };
 }
 
-function createLogger(parent: AzureDebugger, levelOrNamespace: string) {
-  const logger: AzureDebugger = parent.extend(levelOrNamespace) as any;
-  if (levelOrNamespace !== undefined) {
-    logger.level = levelMap[levelOrNamespace as AzureLogLevel];
-  }
+function patchLogMethod(parent: AzureClientLogger, child: AzureClientLogger | AzureDebugger): void {
+  child.log = (...args) => {
```

Comment:
I don't follow the reason for doing this after creating a child logger from AzureLogger above. I suppose it has something to do with stack traces? If a client creates a new logger using `createClientLogger` and wants to further extend it to another namespace (e.g. `azure:storage-blob:, do they also need to patch the log method like this?

Reply:
I believe they would have to, yes. The way `debug` works out of the box doesn't delegate its log method up the chain. So a user wanting to redirect logs to a file or another logging system like winston would have to patch log themselves in different places.

Reply:
`debug` out of the box does copy the log method from the namespace it is extending, but only once at creation. If you redirect after the namespace is set up, then it wouldn't be redirected unless you manually patched it.

So yes, Brian is right that we do this so the user can later redirect AzureLogger.log to somewhere without having to worry about the timing of it.

Reply:
An alternative approach would be to have a setLoggerFunction method that AzureLogger delegates out to (which defaults to debug.log.)

Maybe that would be cleaner?

Reply:
Talked to Brian, the scenario I was missing was when you want to redirect a particular subtree (e.g. all of storage)